app/controllers/whitelist_controller.rb

```diff
@@ -1,31 +1,22 @@
 class WhitelistController < ApplicationController
 
   # http_basic_authenticate_with name: "cs169", password: ENV['PROJECTSCOPE_PASSWORD']
-  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
-  ADMIN_ROLE = "admin"
-  COACH_ROLE = "coach"
-  
+
```

Comment:
Since all controller actions requires admin permission and the else blocks are all the same, it's better to define a function like 
```
def check_if_admin
  unless current_user.is_admin?
     flash[:notice] = ...
     redirect_to .... and return
  end
end
```
and add this line at the top
`before_action :check_if_admin`